hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/http/ReconHttpClient.java

```diff
@@ -0,0 +1,176 @@
+    CloseableHttpClient httpClient = null;
+    try {
+      System.setProperty("sun.security.krb5.debug", "true");
+      System.setProperty("javax.security.auth.useSubjectCredsOnly", "false");
+      Lookup<AuthSchemeProvider> authSchemeRegistry =
+          RegistryBuilder.<AuthSchemeProvider>create()
+              .register(AuthSchemes.SPNEGO, new SPNegoSchemeFactory(true))
+              .build();
+      httpClient = HttpClients.custom()
+          .setSSLHostnameVerifier(
+              NoopHostnameVerifier.INSTANCE)
+          .setSSLContext(new SSLContextBuilder().loadTrustMaterial(
+              null, new TrustStrategy() {
+                public boolean isTrusted(X509Certificate[] arg0, String arg1)
+                    throws CertificateException {
+                  return true;
+                }
+              }).build())
+          .setDefaultAuthSchemeRegistry(authSchemeRegistry).build();
+    } catch (KeyManagementException e) {
+      LOG.error("KeyManagementException in creating http client instance", e);
+    } catch (NoSuchAlgorithmException e) {
+      LOG.error("NoSuchAlgorithmException in creating http client instance", e);
+    } catch (KeyStoreException e) {
+      LOG.error("KeyStoreException in creating http client instance", e);
+    }
+    return httpClient;
+  }
+
+  private static HttpClientContext getHttpClientContext() {
+    HttpClientContext context = HttpClientContext.create();
+    BasicCredentialsProvider credentialsProvider =
+        new BasicCredentialsProvider();
+    // This may seem odd, but specifying 'null' as principal
+    // tells java to use the logged in user's credentials
+    Credentials useJaasCreds = new Credentials() {
+      public String getPassword() {
+        return null;
+      }
+
+      public Principal getUserPrincipal() {
+        return null;
+      }
+    };
+    credentialsProvider.setCredentials(
+        new AuthScope(null, -1, null), useJaasCreds);
+    context.setCredentialsProvider(credentialsProvider);
+    return context;
+  }
+
+  public String sendPost(String host, int port,
+                         String uri, List<NameValuePair> urlParameters)
+      throws IOException {
+    String entityResponse = "";
+    String solrHostBaseQuery = getSolrHostBaseURI(host, port, uri);
+    HttpPost post = new HttpPost(solrHostBaseQuery);
+    LOG.info("Making Http Query to Solr: {} ", solrHostBaseQuery);
+    post.setEntity(new UrlEncodedFormEntity(urlParameters));
+    Instant start = Instant.now();
+    try (CloseableHttpClient httpClient = getCloseableHttpClient();
+         CloseableHttpResponse response = httpClient.execute(post,
+             getHttpClientContext())) {
+      Instant end = Instant.now();
+      int statusCode = response.getStatusLine().getStatusCode();
+      LOG.info("Status Code: {}", statusCode);
+      long duration = Duration.between(start, end).toMillis();
+      LOG.info("Time taken by Solr Query to return: {} milliseconds", duration);
```

Comment:
Please merge this log and above info log into one log message.

Reply:
> Please merge this log and above info log into one log message.
> 
> Maybe we can introduce some metrics about the solr query performance later.

Ok.